File: dnatomrnaranscription.pl
#!/usr/bin/perl -w
# This script performs the transcription of a DNA sequence into an RNA sequence.

# Define the DNA sequence to be transcribed.
# For simplicity, the sequence is hard-coded here, but it can be modified or input dynamically if needed.
$DNA = 'ATATAGATAGATCGCTCGTAGAGATAGCTTCGGCCTCGATAGCTTCTAGGATAATATAGAGATATAGGTTGCGCTAGATAGCTGATAGCTGAGGATCGGCGCGTGATCGGCTCTCTCTCTGGAAAAATAGT';

# Print the original DNA sequence.
print "Here is the DNA to be transcribed:\n\n";
print "$DNA\n";

# Transcription: Convert the DNA sequence to RNA by replacing all occurrences of 'T' with 'U'.
# Explanation:
# - The `=~` operator binds the variable `$RNA` to the substitution operation.
# - The `s/T/U/g` command substitutes 'T' with 'U' globally (i.e., throughout the entire string).
$RNA = $DNA;
$RNA =~ s/T/U/g;

# Print the transcribed RNA sequence.
print "Here is the transcribed RNA:\n\n";
print "$RNA\n";

# Exit the program.
exit;

File: dnatomrnaranscription.py
# DNA to mRNA Transcription

def transcribe_dna_to_mrna(dna_sequence):
    """
    Converts a DNA sequence to an mRNA sequence by replacing T with U.
    
    :param dna_sequence: String representing the DNA sequence
    :return: String representing the transcribed mRNA sequence
    """
    # Replace 'T' with 'U' to transcribe DNA to mRNA
    mrna_sequence = dna_sequence.upper().replace('T', 'U')
    return mrna_sequence

if __name__ == "__main__":
    print("Welcome to the DNA to mRNA Transcription Program!")
    dna_input = input("Enter a DNA sequence: ").strip()
    mrna_output = transcribe_dna_to_mrna(dna_input)
    print(f"Transcribed mRNA sequence: {mrna_output}")
